microbench/src/test/java/io/netty/microbench/http2/Http2FrameWriterBenchmark.java

```diff
@@ -0,0 +1,491 @@
+import io.netty.util.concurrent.EventExecutor;
+
+import java.net.SocketAddress;
+import java.util.Random;
+
+import org.junit.AfterClass;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+
+@State(Scope.Benchmark)
+public class Http2FrameWriterBenchmark extends AbstractMicrobenchmark {
+    private static final EnvironmentParameters NIO_UNPOOLED_PARAMS = new EnvironmentParameters() {
+        @Override
+        public EventLoopGroup newEventLoopGroup() {
+            return new NioEventLoopGroup();
+        }
+
+        @Override
+        public Class<? extends ServerChannel> serverChannelClass() {
+            return NioServerSocketChannel.class;
+        }
+
+        @Override
+        public Class<? extends Channel> clientChannelClass() {
+            return NioSocketChannel.class;
+        }
+
+        @Override
+        public ByteBufAllocator clientAllocator() {
+            return UnpooledByteBufAllocator.DEFAULT;
+        }
+    };
+    private static final EnvironmentParameters NIO_POOLED_PARAMS = new EnvironmentParameters() {
+        @Override
+        public EventLoopGroup newEventLoopGroup() {
+            return new NioEventLoopGroup();
+        }
+
+        @Override
+        public Class<? extends ServerChannel> serverChannelClass() {
+            return NioServerSocketChannel.class;
+        }
+
+        @Override
+        public Class<? extends Channel> clientChannelClass() {
+            return NioSocketChannel.class;
+        }
+
+        @Override
+        public ByteBufAllocator clientAllocator() {
+            return PooledByteBufAllocator.DEFAULT;
+        }
+    };
+    private static final Environment[] environments = { boostrapEnv(NIO_POOLED_PARAMS),
+            boostrapEnv(NIO_UNPOOLED_PARAMS) };
+    private static final byte[][] dataPayload = { createData(256), createData(DEFAULT_MAX_FRAME_SIZE),
+            createData(MAX_FRAME_SIZE_UPPER_BOUND), createData(10 * MAX_FRAME_SIZE_UPPER_BOUND) };
+
+    @Param({ "0", "1", "2", "3" })
+    public int dataPayloadIndex;
+
+    @Param({ "0", "1" })
+    public int envIndex;
+
+    @Param({ "0", "255" })
+    public int padding;
+
+    @AfterClass
+    public static void teardown() {
+        for (Environment env : environments) {
+            try {
+                env.teardown();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Benchmark
+    public void writeData() {
+        doWriteData(environments[envIndex], dataPayload[dataPayloadIndex], padding);
+    }
+
+    private static void doWriteData(Environment env, byte[] data, int padding) {
+        ByteBuf buf = env.clientContext.alloc().buffer(data.length);
+        buf.writeBytes(data);
+        env.clientWriter.writeData(env.clientContext, 3, buf, padding, true, env.clientContext.newPromise());
```

Comment:
Yip I can do that.  However I currently am allocating the input buffer with the allocator which varies based upon the environment setup.  We can just assume that allocation is out of scope for the time being.